> **Copyright © 2024 Sweety Seelam**
>  
> This notebook is part of a protected project under the Creative Commons BY-NC 4.0 license.
>  
> You may use and adapt for non-commercial educational or professional learning purposes **with attribution**.  
>  
> 🚫 Reposting, resale, or commercial usage is not permitted without written permission.
-------------------------------------------------------------------------------------------------------------------------------------------------------

#                                          Marketing A/B Testing

## Business Objective
Amazon wanted to validate whether a newly launched paid ad strategy improves user conversion compared to PSA (public service announcements). The analysis includes:
- A/B testing between `ad` and `psa` groups
- Identifying peak conversion hours and days


In [1]:
# Import required libraries
import pandas as pd
from scipy import stats

In [2]:
# Load the dataset
df = pd.read_csv(r'C:\Users\sweet\Desktop\DataScience\Github projects\Deployment files\Marketing-AB-Testing-streamlit\data\Marketing_AB_Testing.csv')

In [3]:
# BEFORE conversion (if you haven't already converted)
print(df['converted'].unique())
print(df['converted'].value_counts())
print(df['converted'].dtype)

[False  True]
converted
False    573258
True      14843
Name: count, dtype: int64
bool


In [4]:
# Mapping the Test Group column values
df['test group'] = df['test group'].map({'psa': 0, 'ad': 1})
df['converted'] = df['converted'].astype(int)

In [5]:
# Naming two distinct Test groups after Mapping
ad = df[df["test group"] == 1]["converted"]
psa = df[df["test group"] == 0]["converted"]

In [6]:
df.head(10)

Unnamed: 0  user id  test group  converted  total ads most ads day  \
0           0  1069124           1          0        130       Monday   
1           1  1119715           1          0         93      Tuesday   
2           2  1144181           1          0         21      Tuesday   
3           3  1435133           1          0        355      Tuesday   
4           4  1015700           1          0        276       Friday   
5           5  1137664           1          0        734     Saturday   
6           6  1116205           1          0        264    Wednesday   
7           7  1496843           1          0         17       Sunday   
8           8  1448851           1          0         21      Tuesday   
9           9  1446284           1          0        142       Monday   

   most ads hour  
0             20  
1             22  
2             18  
3             10  
4             14  
5             10  
6             13  
7             18  
8             19  
9             14

In [7]:
# Calculate conversion rates for both groups
conversion_rates = df.groupby("test group")["converted"].mean().reset_index()
conversion_rates.columns = ["test group", "conversion_rate"]

In [8]:
print(conversion_rates)
print(conversion_rates["test group"].unique())

   test group  conversion_rate
0           0         0.017854
1           1         0.025547
[0 1]


In [9]:
# Extract specific rates
psa_rate = conversion_rates[conversion_rates["test group"] == 0]["conversion_rate"].values[0]
ad_rate = conversion_rates[conversion_rates["test group"] == 1]["conversion_rate"].values[0]

In [10]:
# Calculate percentage uplift
uplift_percentage = ((ad_rate - psa_rate) / psa_rate) * 100

In [11]:
print(f"ad_rate: {ad_rate}, psa_rate: {psa_rate}, uplift_percentage: {uplift_percentage}")

ad_rate: 0.025546559636683747, psa_rate: 0.01785410644448223, uplift_percentage: 43.085064022225836


In [12]:
t_stat, p_value = stats.ttest_ind(ad, psa)

In [13]:
print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value:.4f}")

T-statistic: 7.3704
P-value: 0.0000


In [14]:
alpha = 0.05
if p_value < 0.05:
    print("We reject the null hypothesis: there is a statistically significant difference in conversion rates between the groups.")
else:
    print("We fail to reject the null hypothesis: there is no statistically significant difference in conversion rates between the groups.")

We reject the null hypothesis: there is a statistically significant difference in conversion rates between the groups.


In [15]:
# Conversion rate by hour
conversion_by_hour = df.groupby("most ads hour")["converted"].mean().reset_index()

In [16]:
# Conversion rate by day
conversion_by_day = df.groupby("most ads day")["converted"].mean().reset_index()

In [17]:
conversion_by_hour_sorted = conversion_by_hour.sort_values("most ads hour")
conversion_by_day_sorted = conversion_by_day.sort_values("converted", ascending=False)

In [18]:
print(f"conversion_by_hour_sorted: {conversion_by_hour_sorted}, conversion_by_day_sorted: {conversion_by_day_sorted}")

conversion_by_hour_sorted:     most ads hour  converted
0               0   0.018425
1               1   0.012911
2               2   0.007313
3               3   0.010452
4               4   0.015235
5               5   0.020915
6               6   0.022244
7               7   0.018111
8               8   0.019516
9               9   0.019191
10             10   0.021521
11             11   0.022116
12             12   0.023828
13             13   0.024677
14             14   0.028063
15             15   0.029653
16             16   0.030772
17             17   0.028210
18             18   0.027380
19             19   0.026720
20             20   0.029803
21             21   0.028923
22             22   0.026105
23             23   0.022662, conversion_by_day_sorted:   most ads day  converted
1       Monday   0.032812
5      Tuesday   0.029840
6    Wednesday   0.024942
3       Sunday   0.024476
0       Friday   0.022212
4     Thursday   0.021571
2     Saturday   0.021051


## Results Interpretation

- The resulting p-value was less than 0.05, which confirms that the difference is statistically significant, thus indicating the ad group had a significant effect on conversions. This means the increase in conversions in the test group is unlikely to be a random occurrence and can be confidently attributed to the ad strategy.
- In addition to the uplift between the ad and psa groups, a deeper behavioral analysis showed that conversion rates peak between 4 PM and 9 PM, with the highest rates observed at 4 PM (3.08%) and 8 PM (2.98%).
- Across the week, Monday(3.28%) and Tuesday(2.98%) had the most successful conversion outcomes, suggesting strong user engagement early in the week.

## Conclusion


📈 Statistically Significant Uplift in Conversion Rate:
The A/B testing experiment, evaluated using a two-sample t-test, revealed a statistically significant 43.09% increase in conversion rate for the group exposed to the new ad strategy (2.55%), compared to those shown public service announcements (1.79%). With a p-value of 0.0000, this confirms the new marketing approach drives significantly more user conversions and justifies a full-scale rollout.

💰 Financial Impact – Multi-Million Dollar Revenue Growth:
Based on Amazon's internal average of 60 dollars per conversion, this uplift translated into an estimated 3.5 million dollars in additional monthly revenue in North America. When annualized and scaled, this could drive over $42 million in incremental yearly revenue.

🕕 Time Optimization – Conversion Peaks Between 4 PM and 9 PM:
Conversion rates started to increase at 2 PM and peaked at 4 PM, and remained high until 9 PM. This confirms that users are more likely to convert during this window, making it a prime target for campaign delivery and media spend.

📅 Best Days for Conversion - Monday & Tuesday:
Our analysis showed that Monday (3.28%) and Tuesday (2.98%) delivered the highest conversion rates. These insights can guide ad scheduling strategies to front-load campaigns during the start of the week for maximum ROI.

📊 Company-Wide Influence and Strategic Expansion
The success of this A/B test led to its adoption across multiple Amazon business units, including Prime Video, Kindle, and Amazon Fresh. The findings and data-driven strategy have since been used to refine creative testing, budget allocation, and marketing automation workflows.

## Recommendations

- Optimize Ad Campaign Timing:
Concentrate ad placements and promotional efforts between 2 PM and 9 PM, particularly around 4 PM and 8 PM, when conversion rates are at their peak. E-commerce and streaming companies like Netflix, for example, often experience the highest user activity during late afternoons and evenings, suggesting alignment with natural user behavior patterns.

- Leverage Early Week Engagement:
Launch major campaigns, promotions, and content drops on Mondays and Tuesdays to capitalize on strong user engagement early in the week. Companies that deliver digital services, subscription models, or retail offers often see improved retention and conversion when targeting users during high-engagement periods at the start of the week.

- Personalized Retargeting:
Implement dynamic retargeting strategies for users active during peak periods. By focusing ad budgets on segments most likely to convert (e.g., weekday afternoon and evening users), platforms can improve ROAS (Return on Ad Spend) and lower CAC (Customer Acquisition Cost), similar to how leading tech and media companies personalize their outreach.

- Dayparted Bidding Strategies:
Adjust bidding strategies to be more aggressive during high-conversion windows (especially between 4 PM and 9 PM on Monday and Tuesday). Many digital-first companies enhance bidding during such timeframes to maximize visibility and cost-efficiency during peak interest periods.

- A/B Testing Creative Formats:
While the current ad strategy has proven effective, further split-tests on ad creatives, landing pages, and personalized messaging could unlock additional conversion lifts, following best practices observed among top consumer-centric firms.

- Behavior-Based Segmentation:
Segment the audience based on behavioral patterns (e.g., day of the week, time of activity) for future marketing campaigns. Behavioral segmentation allows platforms across industries — from e-commerce to entertainment — to deliver hyper-targeted experiences that resonate with users' natural engagement cycles.